### ISSUE TITLE
assertionValue for objectGUID field is wrong if value contains byte 5c

### ISSUE THREAD

**Issue by artxki** [cannatag/ldap3#589]
## Symptoms:
Cannot find some objects in ActiveDirectory when searching by objectGUID. It seems to only affect values that contain `5c` (backslash).

## Example:
```
conn.search(search_filter='(objectGUID=\\7e\\18\\2a\\9c\\60\\5c\\61\\43\\af\\f5\\89\\f5\\e6\\d8\\45\\6d)')
```
Results in:
```
SearchRequest:
 ...
 filter=Filter:
  equalityMatch=EqualityMatch:
   attributeDesc=objectGUID
   assertionValue=0x7e182a9c60acaff589f5e6d8456d
```
That's 2 bytes lost (see the table below).

## Stack:
```
Connection.search  # connection.py
    search_operation  # search.py
        compile_filter()  # search.py
            prepare_filter_for_sending()  # convert.py
```

## Problem:
Seems like `prepare_filter_for_sending()` damages the value, removing 2 bytes from it.

| ~ | value | value.hex() |
| - | - | - |
| receives | b'~\x18*\x9c`\\aC\xaf\xf5\x89\xf5\xe6\xd8Em' | `7e182a9c605c6143aff589f5e6d8456d` |
| outputs | b'~\x18*\x9c`\\aC\xaf\xf5\x89\xf5\xe6\xd8Em' | `7e182a9c60acaff589f5e6d8456d` |

Not sure whether it is a bug in `prepare_filter_for_sending()` itself or in it's usage. 

## Versions
ldap3: PyPI release 2.5.1
Python: 3.6.5

**Comment by artxki** [comment]
`ldap3` `2.4.1` doesn't seem to have this issue. I am using it as a workaround atm.

**Comment by cannatag** [comment]
Hi, cannot replicate this issue, do you still have this problem?

**Comment by cannatag** [comment]
Hi, can you check the code in dev? It should work now.

Thanks,
Giovanni

**Comment by cannatag** [comment]
Fixed in version 2.6.
